crates/sargon/src/profile/mfa/security_structures/security_shield_builder.rs

```diff
@@ -459,6 +459,31 @@ impl SecurityShieldBuilder {
     }
 }
 
+// ================
+// ==== STATIC ====
+// ================
+impl SecurityShieldBuilder {
+    pub fn sorted_factor_sources_for_selection(
+        factor_sources: Vec<FactorSource>,
+    ) -> Vec<FactorSource> {
+        let mut sorted = factor_sources;
+        sorted.sort_by_key(|fs| fs.factor_source_kind().display_order());
+        sorted
+    }
+
+    pub fn selected_factor_sources_status(
+        factor_sources: Vec<FactorSource>,
+    ) -> SelectedFactorSourcesStatus {
+        if factor_sources.is_empty() {
+            SelectedFactorSourcesStatus::Insufficient
```

Comment:
Hmm `1` is never enough for a whole shield. So this must be PER ROLE and which role is it? `Primary` ("Access")?

If so the enum and this method should make it clear that it is for `Primary`. 

(Or if it is dynamic it should be `for_role: RoleKind` as input, and then we should not return just an enum but rather a struct with the status and the role)

Reply:
Post-our sync:
**Delegate this to `MatrixBuilder`**

In MatrixBuilder:
We should probably NOT let this be a STATIC, even if we wont use the internal state of the `MatrixBuilder` **initially** we still might want to in the future. So I would do these changes:
* change it to be on `&self` (i.e. not "static")
* change it to take a `for_role: RoleKind`